### PR TITLE
[DFSM] Move backing up shared storage data to fetch_config recipe

### PR DESCRIPTION
**Pull request by chenwany**
Move backing up shared storage data to fetch_config recipe so it is executed when performing update. Add log message when backing up the recipe
Signed-off-by: chenwany <chenwany@amazon.com>

### Description of changes
* Describe *what* you're changing and *why* you're doing these changes.
* Link to impacted open issues.

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/helpers.rb [hunk 1]
```diff
@@ -629,7 +629,7 @@ def is_slurm_database_updated?
   config["Scheduling"]["SlurmSettings"]["Database"] != previous_config["Scheduling"]["SlurmSettings"]["Database"]
 end
 
-# load cluster configuration file into node object
+# load shared storages data into node object
 def load_shared_storages_mapping
```
Comment (lukeseawalker) [inline libraries/helpers.rb:633]: can this be moved next to 
```
    # load cluster config into node object
    load_cluster_config
````
in the fetch_config method?

Comment (lukeseawalker) [inline libraries/helpers.rb:633]: can this be moved in the fetch_config now?

Comment (chenwany) [reply libraries/helpers.rb:633]: Changed

**Review by lukeseawalker — DISMISSED** [review]

File: libraries/helpers.rb (hunk 1 above)
Comment (chenwany) [reply libraries/helpers.rb:633]: After making this change, I run a test and it didn't work. The node attribute is not loaded later after I move load_shared_storage_data to fetch_config. I am going to keep it in its original place and let's have a further discussion on where is better to keep the function

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**